packages/flutter/test/cupertino/scaffold_test.dart

```diff
@@ -53,6 +54,72 @@ void main() {
     expect(tester.getRect(find.byType(Container)), const Rect.fromLTRB(0, 44, 800, 600));
   });
 
+  testWidgets('dark mode and obstruction work', (WidgetTester tester) async {
+    const Color dynamicColor = CupertinoDynamicColor.withBrightness(
+      color: Color(0xFFF8F8F8),
+      darkColor: Color(0xEE333333),
+    );
+
+    const CupertinoDynamicColor backgroundColor = CupertinoDynamicColor.withBrightness(
+      color: Color(0xFFFFFFFF),
+      darkColor: Color(0xFF000000),
+    );
+
+    BuildContext childContext;
+    await tester.pumpWidget(Directionality(
+      textDirection: TextDirection.ltr,
+      child: MediaQuery(
+        data: const MediaQueryData(
+          platformBrightness: Brightness.light,
+          viewInsets: EdgeInsets.only(top: 20),
+        ),
+        child: CupertinoPageScaffold(
+          backgroundColor: backgroundColor,
+          navigationBar: const CupertinoNavigationBar(
+            middle: Text('Title'),
+            backgroundColor: dynamicColor,
+          ),
+          child: Builder(
+            builder: (BuildContext context) {
+              childContext = context;
+              return Container();
+            },
+          ),
+        ),
+      ),
+    ));
+
+    expect(MediaQuery.of(childContext).padding.top, 0);
+    expect(find.byType(CupertinoPageScaffold), paints..rect(color: backgroundColor.color));
+
+    await tester.pumpWidget(Directionality(
```

Comment:
What do you think about creating a function that returns this widget tree and accepts the brightness as a parameter?  I know it's super nit picky, but I think that it would make it easier to understand at a glance that brightness is the only thing changing between these two pumpWidget calls.